web/src/main/kotlin/com/example/ajaxproject/service/GroupChatServiceImpl.kt

```diff
@@ -30,107 +34,94 @@ class GroupChatServiceImpl (
         private val logger: Logger = LoggerFactory.getLogger(GroupChatServiceImpl::class.java)
     }
 
-    override fun createGroupRoom(createChatDto: CreateChatDto): GroupChatRoom {
-
-        val user = userService.getUserById(createChatDto.adminId)
-
-        val groupChatRoom = GroupChatRoom(
-            id = ObjectId().toHexString(),
-            chatName = createChatDto.chatName,
-            adminId = user.id,
-            chatMembers = listOf(user)
-        )
-
-        groupChatRoom.chatMembers + user
-
-        groupChatRoomRepository.save(groupChatRoom)
-
-        logger.info("Group chat room created successfully")
-
-        return groupChatRoom
-
+    override fun createGroupRoom(createChatDto: CreateChatDTO): Mono<GroupChatRoom> {
+        return userService.getById(createChatDto.adminId)
+            .flatMap { user ->
+                val groupChatRoom = GroupChatRoom(
+                    id = ObjectId().toHexString(),
+                    chatName = createChatDto.chatName,
+                    adminId = user.id,
+                    chatMembers = listOf(user)
+                )
+                groupChatRoomRepository.save(groupChatRoom)
+                    .doOnSuccess { logger.info("Group chat room created successfully") }
+            }
     }
 
-    override fun getAllChatMembers(chatId: String): List<User> {
-
-        val chatMembers = groupChatRoomRepository.findChatRoom(chatId).chatMembers
-
-        logger.info("Retrieved chat members for chat ID: {}" , chatId)
-
-        return chatMembers
+    override fun getAllChatMembers(chatId: String): Mono<List<User>> {
```

Comment:
Maybe we should return `Flux<User>`?

Reply:
In this case code will look more better. 

```suggestion
override fun getAllChatMembers(chatId: String): Flux<User> {
    return groupChatRoomRepository.findChatRoom(chatId)
        .flatMapMany { chatRoom ->
            Flux.fromIterable(chatRoom.chatMembers)
        }
}
```

Reply:
But see my answer to the previous conversation

Reply:
which one comment? I cannot see the order of your replies